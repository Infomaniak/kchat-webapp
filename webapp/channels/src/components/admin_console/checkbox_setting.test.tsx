// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {renderWithContext, screen, fireEvent} from 'tests/react_testing_utils';

import CheckboxSetting from './checkbox_setting';

describe('components/admin_console/CheckboxSetting', () => {
    test('should match snapshot', () => {
        const onChange = jest.fn();
        const {container} = renderWithContext(
            <CheckboxSetting
                id='string.id'
                label='some label'
                defaultChecked={false}
                onChange={onChange}
                setByEnv={false}
                disabled={false}
            />,
        );
        const checkbox: HTMLInputElement = screen.getByRole('checkbox');
        expect(checkbox).toBeVisible();
        expect(checkbox).toHaveProperty('type', 'checkbox');
        expect(container).toMatchSnapshot();
    });

    test('onChange', () => {
        const onChange = jest.fn();
        renderWithContext(
            <CheckboxSetting
                id='string.id'
                label='some label'
                defaultChecked={false}
                onChange={onChange}
                setByEnv={false}
                disabled={false}
            />,
        );
        const checkbox: HTMLInputElement = screen.getByRole('checkbox');
        expect(checkbox).not.toBeChecked();

        fireEvent.click(checkbox);

        expect(onChange).toHaveBeenCalledTimes(1);
        expect(onChange).toHaveBeenCalledWith('string.id', true);
        expect(checkbox).toBeChecked();
    });
});
