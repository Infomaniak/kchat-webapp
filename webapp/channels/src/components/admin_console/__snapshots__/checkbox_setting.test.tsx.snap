// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/admin_console/CheckboxSetting should match snapshot 1`] = `
<div>
  <div
    class="form-group"
    data-testid="string.id"
  >
    <div
      class="col-sm-12"
    >
      <a
        id="string.id"
      />
      <label
        class="checkbox-inline"
      >
        <input
          data-testid="string.id"
          id="string.id"
          name="string.id"
          type="checkbox"
        />
        some label
      </label>
      <div
        class="help-text"
        data-testid="string.idhelp-text"
      />
    </div>
  </div>
</div>
`;
